lib/oli_web/live/datashop/analytics_live.ex

```diff
@@ -0,0 +1,319 @@
+        datashop_export_timestamp: datashop_export_timestamp,
+        selected_sections: selected_sections
+      )
+
+    {:ok, socket}
+  end
+
+  defp determine_total(sections) do
+    case(sections) do
+      [] -> 0
+      [hd | _] -> hd.total_count
+    end
+  end
+
+  def handle_params(params, _, socket) do
+    table_model =
+      OliWeb.Common.Table.SortableTableModel.update_from_params(
+        socket.assigns.table_model,
+        params
+      )
+
+    offset = get_int_param(params, "offset", 0)
+    text_search = get_param(params, "text_search", "")
+    project = socket.assigns.project
+
+    sections =
+      Browse.browse_sections(
+        %Paging{offset: offset, limit: @limit},
+        %Sorting{direction: table_model.sort_order, field: table_model.sort_by_spec.name},
+        %BrowseOptions{
+          project_id: project.id,
+          blueprint_id: nil,
+          text_search: text_search,
+          active_today: false,
+          filter_status: :active,
+          filter_type: nil,
+          institution_id: nil
+        }
+      )
+
+    displayed_sections = Enum.map(sections, & &1.id)
+
+    selected_sections =
+      MapSet.filter(socket.assigns.selected_sections, fn s -> s in displayed_sections end)
+
+    table_model =
+      table_model
+      |> Map.put(:rows, sections)
+      |> Map.update!(:data, fn data ->
+        %{data | selected_sections: selected_sections}
+      end)
+
+    total_count = determine_total(sections)
+
+    {:noreply,
+     assign(socket,
+       offset: offset,
+       table_model: table_model,
+       total_count: total_count,
+       text_search: text_search,
+       selected_sections: selected_sections
+     )}
+  end
+
+  attr(:breadcrumbs, :any, default: [Breadcrumb.new(%{full_title: "Datashop Analytics"})])
+  attr(:title, :string, default: "Datashop Analytics")
+
+  attr(:tabel_model, :map)
+  attr(:total_count, :integer, default: 0)
+  attr(:offset, :integer, default: 0)
+  attr(:limit, :integer, default: @limit)
+  attr(:text_search, :string, default: "")
+  attr(:selected_sections, :map, default: MapSet.new([]))
+
+  def render(assigns) do
+    ~H"""
+    <Modal.modal
+      id="generate_datashop_export_modal"
+      class="w-1/3"
+      on_confirm={
+        JS.push("generate_datashop_snapshot")
+        |> Modal.hide_modal("generate_datashop_export_modal")
+      }
+    >
+      <:title>Generate Datashop Export</:title>
+      <div class="flex flex-col items-center justify-center text-center p-4">
+        <div class="mb-4 p-4 border border-orange-300 dark:border-orange-700 bg-orange-100 dark:bg-orange-800 rounded-lg shadow-md">
+          <i class="fas fa-exclamation-triangle text-orange-500 dark:text-orange-200 text-xl mr-2">
+          </i>
+          <span class="text-orange-600 dark:text-orange-300">
+            DataShop Download is an unstable feature at the moment as the memory demands that it places on the system can cause the server to crash.
```

Comment:
We can remove this warning message about the being unstable, now that the previous round of the "streaming" based impl has landed.